data/models.py

```diff
@@ -17,17 +18,155 @@ class DDSUserCredential(models.Model):
     """
     DDS Credentials for bespin users
     """
+    endpoint = models.ForeignKey(DDSEndpoint, on_delete=models.CASCADE, null=False)
     user = models.OneToOneField(settings.AUTH_USER_MODEL, null=False)
     token = models.CharField(max_length=32, blank=False, unique=True)
 
 
-class DDSResource(models.Model):
-    owner = models.ForeignKey(settings.AUTH_USER_MODEL, null=False)
-    project_id = models.UUIDField(blank=False)
-    path = models.TextField(null=False)
+class Workflow(models.Model):
+    """
+    Name of a workflow that will apply some processing to some data.
+    """
+    name = models.CharField(max_length=255, blank=False)
+
+    def __unicode__(self):
+        return self.name
 
-    class Meta:
-        unique_together = (
-            ('owner', 'project_id', 'path'),
-        )
 
+class WorkflowVersion(models.Model):
+    """
+    Specific version of a Workflow.
+    """
+    workflow = models.ForeignKey(Workflow, on_delete=models.CASCADE, related_name='versions')
+    object_name = models.CharField(max_length=255, null=True)
```

Comment:
Unclear from the context what this parameter is for - comment would be useful
Also, if `#main` is a reasonable default, would make sense to mark it as a default.